orchestration/src/main/java/com/sap/ai/sdk/orchestration/OrchestrationModuleConfig.java

```diff
@@ -0,0 +1,51 @@
+package com.sap.ai.sdk.orchestration;
+
+import com.sap.ai.sdk.orchestration.client.model.FilteringModuleConfig;
+import com.sap.ai.sdk.orchestration.client.model.LLMModuleConfig;
+import com.sap.ai.sdk.orchestration.client.model.MaskingModuleConfig;
+import com.sap.ai.sdk.orchestration.client.model.TemplatingModuleConfig;
+import javax.annotation.Nullable;
+import lombok.AccessLevel;
+import lombok.AllArgsConstructor;
+import lombok.NoArgsConstructor;
+import lombok.Value;
+import lombok.With;
+
+/**
+ * Represents the configuration for the orchestration service. Allows for configuring the different
+ * modules of the orchestration service via a fluent API.
+ *
+ * <p>The orchestration pipeline combines different modules into a single execution flow where the
+ * output of one module serves as input for the next. The pipeline consists of the following
+ * modules:
+ *
+ * <ul>
+ *   <li>LLM Config (Mandatory)
+ *   <li>Templating (Optional)
+ *   <li>Data Masking (Optional)
+ *   <li>Input Content Filtering (Optional)
+ *   <li>Output Content Filtering (Optional)
+ * </ul>
+ */
+@Value
+@With
+@AllArgsConstructor(access = AccessLevel.PRIVATE)
+@NoArgsConstructor(force = true)
+public class OrchestrationModuleConfig {
```

Comment:
`@With` leads to dead stores as outlined by SonarLint.
Plus the lack of proper builder mean you throw `IllegalStateException` in the `ModuleConfigFactory`.